core/base/inc/Rtypes.h

```diff
@@ -284,98 +285,111 @@ private:          \
+   }
 
 #define ClassDef(name,id)                            \
    _ClassDefOutline_(name,id,virtual,)               \
-   static int DeclFileLine() { return __LINE__; }
+   /** \cond HIDDEN_SYMBOLS */ static int DeclFileLine() { return __LINE__; } /** \endcond */
 
 #define ClassDefOverride(name,id)                    \
    _ClassDefOutline_(name,id,,override)              \
-   static int DeclFileLine() { return __LINE__; }
+   /** \cond HIDDEN_SYMBOLS */ static int DeclFileLine() { return __LINE__; } /** \endcond */
 
 #define ClassDefNV(name,id)                          \
    _ClassDefOutline_(name,id,,)                      \
-   static int DeclFileLine() { return __LINE__; }
+   /** \cond HIDDEN_SYMBOLS */ static int DeclFileLine() { return __LINE__; } /** \endcond */
 
 #define ClassDefInline(name,id)                      \
    _ClassDefInline_(name,id,virtual,)                \
-   static int DeclFileLine() { return __LINE__; }
+   /** \cond HIDDEN_SYMBOLS */ static int DeclFileLine() { return __LINE__; } /** \endcond */
 
 #define ClassDefInlineOverride(name,id)              \
    _ClassDefInline_(name,id,,override)               \
-   static int DeclFileLine() { return __LINE__; }
+   /** \cond HIDDEN_SYMBOLS */ static int DeclFileLine() { return __LINE__; } /** \endcond */
 
 #define ClassDefInlineNV(name,id)                    \
    _ClassDefInline_(name,id,,)                       \
-   static int DeclFileLine() { return __LINE__; }
+   /** \cond HIDDEN_SYMBOLS */ static int DeclFileLine() { return __LINE__; } /** \endcond */
 
 //#define _ClassDefInterp_(name,id) ClassDefInline(name,id)
 
 #define R__UseDummy(name) \
    class _NAME2_(name,_c) { public: _NAME2_(name,_c)() { if (name) { } } }
 
-
-#define ClassImpUnique(name,key) \
-   namespace ROOT { \
-      TGenericClassInfo *GenerateInitInstance(const name*); \
-      namespace { \
-         static int _R__UNIQUE_(_NAME2_(R__dummyint,key)) __attribute__((unused)) = \
-            GenerateInitInstance((name*)0x0)->SetImplFile(__FILE__, __LINE__); \
-         R__UseDummy(_R__UNIQUE_(_NAME2_(R__dummyint,key))); \
-      } \
+#define ClassImpUnique(name,key)                                                                        \
+   namespace ROOT {                                                                                     \
+      /** \cond HIDDEN_SYMBOLS */ TGenericClassInfo *GenerateInitInstance(const name*); /** \endcond */ \
+      namespace {                                                                                       \
+         /** \cond HIDDEN_SYMBOLS */                                                                    \
+         static int _R__UNIQUE_(_NAME2_(R__dummyint,key)) __attribute__((unused)) =                     \
+            GenerateInitInstance((name*)0x0)->SetImplFile(__FILE__, __LINE__);  /** \endcond */         \
+         R__UseDummy(_R__UNIQUE_(_NAME2_(R__dummyint,key)));                                            \
```

Comment:
This line declare a variable that probably should also be hidden.

Reply:
Thanks for the hint. I just checked, and we have the anonymous namespace documentation disabled, see:
https://www.doxygen.nl/manual/config.html#cfg_extract_anon_nspaces

Question: do we want, in general, to add documentation for anonymous namespaces? If yes, I can turn the flag to true in the Doxyfile and mark the variable you mentioned as HIDDEN_SYMBOLS. If not, I can remove the HIDDEN_SYMBOLS flag at line 367, as it would be redundant.

Reply:
> do we want, in general, to add documentation for anonymous namespaces? 

No :)

Reply:
Changed